### DISCUSSION TITLE
Support for SV structures

### DISCUSSION THREAD

**Discussion by TomO700** [cocotb/cocotb#3006] [Q&A]
Hi,

My dut has a sv packed struct like what's below.

```
typedef struct packed{
        logic [63:0]   data;
        logic   [7:0]    addr;
        logic               wen;
        logic               ren;
        logic               en;
    }mem_s;
    
```
Is it possible to assign data through cocotb to each element of the structure separately? As I noticed the only straight forward way was to consider the structure as a single array and assign data, which actually defeats the convenience of using a struct.

Thank you

**Comment by grahamcx** [comment]
I've just raised an issue related to this (https://github.com/cocotb/cocotb/issues/3039). I think the support for structs is simulator-specific so I think your request is a request for the simulator developer as cocotb seems to do the 'right' thing if the simulator fully supports structs. 

I've been getting cocotb running using both icarus and vcs and vcs allows you to assign struct members as you would expect via cocotb `dut.struct.member.value = 1` etc. Whereas icarus only has the option to consider the struct as a packed bit vector.
